back/boxtribute_server/business_logic/warehouse/location/crud.py

```diff
@@ -0,0 +1,94 @@
+from ....db import db
+from ....enums import BoxState
+from ....models.definitions.location import Location
+from ....models.utils import (
+    save_creation_to_history,
+    save_deletion_to_history,
+    save_update_to_history,
+    utcnow,
+)
+
+
+@save_creation_to_history
+def create_location(
+    *,
+    name,
+    base_id,
+    box_state=BoxState.InStock,
+    is_shop=False,
+    is_stockroom=False,
+    description="",
+    user_id,
+):
+    """Insert information for a new Location in the database."""
+    now = utcnow()
+    return Location.create(
+        name=name,
+        base=base_id,
+        box_state=box_state,
+        description=description,
+        is_stockroom=is_stockroom,
+        is_shop=is_shop,
+        is_donated=box_state == BoxState.Donated,
+        is_lost=box_state == BoxState.Lost,
+        is_scrap=box_state == BoxState.Scrap,
+        created_on=now,
+        created_by=user_id,
+    )
+
+
+@save_update_to_history(
+    fields=[
+        Location.name,
+        Location.description,
+        Location.box_state,
+        Location.is_stockroom,
+        Location.is_shop,
+        Location.is_donated,
+        Location.is_lost,
+        Location.is_scrap,
+    ],
+)
+def update_location(
+    *,
+    id,  # required for save_update_to_history
+    # location,
+    name=None,
+    description=None,
+    box_state=None,
+    is_stockroom=None,
+    is_shop=None,
+    user_id,
+):
+    """Look up an existing Location given an ID, and update all requested fields.
+    Insert timestamp for modification and return the location.
+    """
+    location = Location.get_by_id(id)
+
+    if name is not None:
+        location.name = name
+    if description is not None:
+        location.description = description
+    if box_state is not None:
+        location.box_state = box_state
+        location.is_donated = box_state == BoxState.Donated
+        location.is_lost = box_state == BoxState.Lost
```

Comment:
take out lost and scrap

Reply:
Does it make sense to keep it for consistency until the `is_lost` and `is_scrap` fields are removed from the locations table?